seshdash/views.py

```diff
@@ -615,3 +616,77 @@ def get_latest_bom_data(request):
     latest_bom_data.append({"item": "Recent Contact", "value": get_timesince(latest_bom.time)})
 
     return HttpResponse(json.dumps(latest_bom_data))
+
+
+@login_required
+def historical_data(request):
+    
+    # If ajax request
+    if request.method == 'POST': 
+        sort_value = request.POST.get('sort_value', '')
+        historical_data = get_historical_dict(column=sort_value)
+        return HttpResponse(json.dumps(historical_data))
+   
+    # On page load
+    else:
+        sites = get_objects_for_user(request.user, 'seshdash.view_Sesh_Site')
+        active_site = sites[0]
+        context_dict = {}
+        context_dict['sites'] = sites
+        context_dict['site_id'] = 0
+        context_dict['active_site'] = active_site
+        return render(request, 'seshdash/historical-data.html', context_dict);
+
+
+def get_alerts_for_year(site):
+    """ Returns a json object containing the number of alerts for a year range """
+ 
+    # NOTE: When using datetime object date_range does not include the last date
+    now = datetime.datetime.now()
+    year_before = now - relativedelta(years=1)
+    number_of_alerts = Sesh_Alert.objects.filter(date__range=[
```

Comment:
you could use simply the `year` filter instead of a date range here and let django handle this for you.